engine/compiler/compiler.go

```diff
@@ -273,24 +291,58 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
 		// create the step
 		spec.Steps = append(spec.Steps, &engine.Step{
 			Step: lespec.Step{
-				Command:    command,
-				Detach:     src.Detach,
-				Envs:       stepEnv,
-				Entrypoint: entrypoint,
-				Files:      files,
-				ID:         stepID,
-				Image:      src.Image,
-				Name:       src.Name,
-				Privileged: src.Image != "", // all steps that use images, run in privileged mode
-				Secrets:    stepSecrets,
-				WorkingDir: workingDir,
-				Volumes:    volumeMounts,
+				Command:      command,
+				Detach:       src.Detach,
+				Envs:         stepEnv,
+				Entrypoint:   entrypoint,
+				Files:        files,
+				ID:           stepID,
+				Image:        src.Image,
+				Name:         src.Name,
+				PortBindings: src.PortBindings,
+				Privileged:   src.Image != "", // all steps that use images, run in privileged mode
+				Secrets:      stepSecrets,
+				WorkingDir:   workingDir,
+				Volumes:      volumeMounts,
 			},
 			DependsOn: src.DependsOn,
 			RunPolicy: runPolicy,
 		})
 	}
 
+	// get registry credentials from registry plugins
+	creds, err := c.Registry.List(ctx, &registry.Request{
+		Repo:  args.Repo,
+		Build: args.Build,
+	})
+	if err != nil { //nolint:staticcheck
+		// TODO (bradrydzewski) return an error to the caller if the provider returns an error.
```

Comment:
Doesn't seem much point to this statement?

Reply:
See the comment above... This particular code section is copied from the docker runner along with the "to do" comment.